grove/pyqaoa/README.md

```diff
@@ -1,61 +1,67 @@
-# pyqaoa
-A python implementation of the Quantum Approximate Optimization Algorithm using
-pyQuil and the Rigetti Forest.
+# pyQAOA
+A Python implementation of the Quantum Approximate Optimization Algorithm (QAOA) using
+pyQuil and the Rigetti Forest SDK.
 
 ## Structure
 
 `qaoa.py` contains the base QAOA class and routines for finding the optimal
-rotation angles via the variational-quantum-eigensolver method, state
-preparation methods, storing results, and utilities for probabilities and
-collecting bitstrings after a state preparation.
+rotation angles via the variational-quantum-eigensolver (VQE) method, preparing states,
+and storing results as well as utilities for probabilities and collecting bitstrings after a
+state preparation.
 
-`maxcut_qaoa.py` takes a graph defined with either NetworkX or a list of node
-pairs and implements the cost function for MAX-CUT problems.
+`maxcut_qaoa.py` takes a graph defined with either NetworkX or a list of edges expressed as tuples
+and implements the cost function for max-cut problems.
 
-`numberpartiition_qaoa.py` takes a list of numbers and set sup a QAOA instance
+`numberpartiition_qaoa.py` takes a list of numbers and sets up a QAOA instance
 for determining the equal biparitioning of the list.
 
 ## Run
  
-The simplest way to interact with the QAOA library is through the methods provided for each problem instance.  For example, to run max cut import `maxcut_qaoa` from `maxcut_qaoa.py` and pass graph to the script. 
+The simplest way to interact with the QAOA library is through the methods provided for
+each problem instance.  For example, to run max-cut import `maxcut_qaoa` from `maxcut_qaoa.py`
+and pass a graph to the script.
 This function will return a QAOA instance.  Calling `get_angles()` on the instance will
-start the variational-quantum-eigensolver loop in order to find  the beta, gamma angles.
+start the variational-quantum-eigensolver (VQE) loop in order to find the optimal beta and gamma
+angles.
 
-## Examples using qaoa
+## Examples using QAOA
 
 ```
 import numpy as np
+from pyquil.api import WavefunctionSimulator
+
 from grove.pyqaoa.maxcut_qaoa import maxcut_qaoa
-import pyquil.api as api
-qvm_connection = api.QVMConnection()
 ```
 
 ```
+n_angles = 30
+n_qubits = 4
+steps = 2
 square_ring = [(0,1),(1,2),(2,3),(3,0)]
-steps = 2; n_qubits = 4
-betas = np.random.uniform(0, np.pi, p); gammas = np.random.uniform(0, 2*np.pi, p)
+
+betas = np.random.uniform(0, np.pi, n_angles); gammas = np.random.uniform(0, 2 * np.pi, n_angles)
```

Comment:
hmm I think `p` (the previous value) was supposed to be the number of steps, usually denoted `p`, which is probably `steps`

Reply:
I'm curious how inputting 30 angles and requesting 2 steps seems to work without incident

Reply:
It turns out `betas` and `gammas` are never used. Worse still, running the code in the `README.md` didn't converge to the expected solutions because we weren't parameterizing with respect to the optimized beta and gamma values. I just pushed a fix.